include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -460,73 +460,26 @@ class is_operational_impl
     {
-        for (auto i = 0u; i < input_bdl_wires.size(); i++)
-        {
-            if (input_bdl_wires[input_bdl_wires.size() - 1 - i].port.dir == port_direction::SOUTH)
-            {
-                if ((current_input_index & (uint64_t{1ull} << i)) != 0ull)
-                {
-                    for (const auto& bdl : input_bdl_wires[input_bdl_wires.size() - 1 - i].pairs)
-                    {
-                        if (bdl.type == sidb_technology::INPUT)
-                        {
-                            continue;
-                        }
-                        if (!encodes_bit_one(ground_state, bdl, input_bdl_wires[input_bdl_wires.size() - 1 - i].port))
-                        {
-                            return true;
-                        }
-                    }
-                }
-                else
-                {
-                    for (const auto& bdl : input_bdl_wires[input_bdl_wires.size() - 1 - i].pairs)
-                    {
-                        if (bdl.type == sidb_technology::INPUT)
-                        {
-                            continue;
-                        }
-                        if (!encodes_bit_zero(ground_state, bdl, input_bdl_wires[input_bdl_wires.size() - 1 - i].port))
-                        {
-                            return true;
-                        }
-                    }
-                }
-            }
-            else if (input_bdl_wires[input_bdl_wires.size() - 1 - i].port.dir == port_direction::NORTH)
-            {
-                if ((current_input_index & (uint64_t{1ull} << i)) != 0ull)
-                {
-                    for (const auto& bdl : input_bdl_wires[input_bdl_wires.size() - 1 - i].pairs)
-                    {
-                        if (bdl.type == sidb_technology::INPUT)
-                        {
-                            continue;
-                        }
-                        if (!encodes_bit_one(ground_state, bdl, input_bdl_wires[input_bdl_wires.size() - 1 - i].port))
-                        {
-                            return true;
-                        }
-                    }
-                }
-                else
-                {
-                    for (const auto& bdl : input_bdl_wires[input_bdl_wires.size() - 1 - i].pairs)
-                    {
-                        if (bdl.type == sidb_technology::INPUT)
-                        {
-                            continue;
-                        }
-                        if (!encodes_bit_zero(ground_state, bdl, input_bdl_wires[input_bdl_wires.size() - 1 - i].port))
-                        {
-                            return true;
-                        }
-                    }
-                }
-            }
-        }
-
-        return false;
+        return std::any_of(input_bdl_wires.rbegin(), input_bdl_wires.rend(),
+                           [&, i = 0u](const auto& wire) mutable
```

Comment:
Global capture `[&]` should be avoided so as not to unintentionally capture variables. Rather, capture by name `[&ground_state, &current_input_index]`

Reply:
you are right! sorry for overseeing this.